spec/order_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'order'
+
+describe Order do
+  
+  let(:pizza) { double(name: 'pizza', price: 7) }
+  let(:pasta) { double(name: 'pasta', price: 15) }
+  let(:olives) { double(name: 'olives', price: 2) }
+  let(:menu_list) { [pizza, pasta, olives] }
+  let(:order) { Order.new(menu_list) }
+  
+  it 'has an empty order_list when instantiated' do
+    expect(order).to have_attributes(order_list: [])
```

Comment:
This is testing state, not behaviour. You want to avoid tests that peek inside a class to see what the state of an instance variable is. Better to have one test that checks whether you can successfully add a dish, and then another test that uses use another method to check that both have worked.

Think back to the Piggy Bank example from Week 1. You can't see the coins inside the piggy bank without breaking it, so you can't test whether once you put a coin in, the piggy bank now contains that coin. But you _can_ test the behaviour of the class, ie. once you .shake it, it now goes "cling!" whereas before it returned nothing.

Similarly, you shouldn't be testing that calling methods alters the internal state of your order_list, you should only be testing the inputs and outputs of the class.